### PR TITLE
connect comments reducer to API

### PR DESCRIPTION
**Pull request by aliceriot**
#### What are the relevant tickets?

part of #18 

#### What's this PR do?

This adds support for creating comments (replies to a post) and replies to comments. 

#### How should this be manually tested?

You could open up `PostPage.js` and add a line like `actions = window.actions`, and then try calling `actions.comments.post(postId, commentText, ?commentId)` to try things out.

### CODE REVIEW

File: static/js/lib/api_test.js [hunk 1]
```diff
@@ -68,5 +71,45 @@ describe("api", function() {
...
       })
     })
+
+    it("gets comments for a post", () => {
+      const post = makePost()
+      const tree = makeCommentTree(post)
+      fetchStub.returns(Promise.resolve(tree))
+
+      return getComments(post.id).then(resp => {
+        assert.deepEqual(tree, resp)
```
Comment (rhysyngsun) [inline static/js/lib/api_test.js]: Could you reverse the args? `expected` is the second.

Comment (aliceriot) [inline static/js/lib/api_test.js]: haha, I never can keep that straight

Comment (aliceriot) [inline static/js/lib/api_test.js]: I look up the argument order for `assert.include` literally every time I use it

File: static/js/reducers/comments.js [hunk 2]
```diff
@@ -1,18 +1,19 @@
...
 export const commentsEndpoint = {
   name:              "comments",
-  verbs:             [GET],
-  getFunc:           async (post: Post) => makeCommentTree(post),
+  verbs:             [GET, POST],
+  getFunc:           (post: Post) => api.getComments(post.id),
   initialState:      { ...INITIAL_STATE, data: new Map() },
   getSuccessHandler: (comments: Array<Comment>, data: Map<string, Array<Comment>>) => {
     let update = new Map(data)
     update.set(comments[0].post_id, comments)
```
Comment (rhysyngsun) [inline static/js/reducers/comments.js]: This line errors if there are no comments on the post.

File: static/js/reducers/comments.js [hunk 3]
```diff
@@ -1,18 +1,19 @@
...
     return update
-  }
+  },
+  postFunc: (post: Post, comment: string, commentId: ?string) => api.createComment(post.id, comment, commentId)
```
Comment (rhysyngsun) [inline static/js/reducers/comments.js]: Figured out why I was getting the 403, this code accepts a `Post` object, but the test code in your PR description says to pass just the id. Code works with the `Post`, but just want to confirm which one is the preferred api here. This code is also subject to the other error I posted.

**Comment by rhysyngsun** [comment]
Both `dispatch(actions.comments.post({id:'1'}, 'comment'))` and `dispatch(actions.comments.post({id:'1'}, 'comment', '2'))` give me the above error.

**Comment by aliceriot** [comment]
ok! that error should be addressed now.

**Comment by rhysyngsun** [comment]
Should the new comment from the post to the comments api be reflected in the UI afterwards? I'm not seeing it if so.

**Comment by aliceriot** [comment]
oh that would make sense :woman_facepalming:

**Comment by aliceriot** [comment]
OK I think all the issues should be fixed now. I'm going to leave off for now handling the merging-in of updated data since it's somewhat nontrivial - I think for now it may be easier to just refresh the whole comment tree instead of trying to correctly merge the updated comment into the data we already have.

**Comment by aliceriot** [comment]
(updated data after you issue a `POST` request, that is)

**Pull request merged**